models/activities/action.go

```diff
@@ -450,17 +451,50 @@ func GetFeeds(ctx context.Context, opts GetFeedsOptions) (ActionList, int64, err
 		return nil, 0, err
 	}
 
-	sess := db.GetEngine(ctx).Where(cond).
-		Select("`action`.*"). // this line will avoid select other joined table's columns
-		Join("INNER", "repository", "`repository`.id = `action`.repo_id")
+	actions := make([]*Action, 0, opts.PageSize)
+	var count int64
 
-	opts.SetDefaultValues()
-	sess = db.SetSessionPagination(sess, &opts)
+	if opts.Page < 10 { // TODO: why it's 10 but other values? It's an experience value.
```

Comment:
What I haven't understood so far:
Why `Page < 10`?
Shouldn't it be `PageSize < 10`? 
Why do we switch the algorithm after the first ~100 entries?

Reply:
> Why do we switch the algorithm after the first ~100 entries?

Take a look into  #31752

Reply:
My question is: 
If I understand correctly, it is mainly caused by the large `OFFSET`? The bigger it is, the slower the query is?
According to the image in the link mentioned in https://github.com/go-gitea/gitea/issues/31752#issuecomment-2263754818

If I'm wrong, then no problem.

Then, page = offset (the limit of using old or new code) / pagesize
And page size is not a const, so the value of page will be dynamic?

Reply:
Yes. But 10 page is an experience value. If we want to change it to an OFFSET, it's also an experience value. I don't think we need accurate for an experience value.

Reply:
If 10 is too low, than take 20 oder 30. ;)

Reply:
It is ok. But maybe more details in comment is better, or it is hard or takes time to trace where these codes and  experience value come from in sometimes. e.g. the whole file moved to another place.